tony-core/src/main/java/com/linkedin/tony/tensorflow/TonySession.java

```diff
@@ -256,8 +253,10 @@ public void onTaskCompleted(String jobName, String jobIndex, int exitCode) {
           if (isChief(jobName, jobIndex)) {
             trainingFinished = true;
           }
+          task.getTaskInfo().setState(TaskStatus.FAILED);
           setFinalStatus(FinalApplicationStatus.FAILED, "Exit status: " + exitCode);
         }
+        task.getTaskInfo().setState(TaskStatus.SUCCEEDED);
```

Comment:
This doesn't seem right? Should this be in an else {} ?

Reply:
right..